### DISCUSSION TITLE
Compile a nanobind extension as part of a larger python package

### DISCUSSION THREAD

**Discussion by OasisArtisan** [wjakob/nanobind#799] [Q&A]
Hello, I have a python project where I have defined a package with multiple submodules.
I want to compile the nanobind extension such that it can be imported using the same project structure.

Ex: Assume I already have the following python package structure.
```
--fruits
----__init__.py
----apple
------__init__.py
------mod1.py
------mod2.py
------nanobindmod1.???
----orange
------__init__.py
------mod3.py
------mod4.py
------nanobindmod2.???
```
Where I can import as `import fruits.apple.mod1`
I want to be able to import `import fruits.apple.nanobindmod1` and `import fruits.orange.nanobindmod2` even though there is a single nanobind extension that was compiled.

Basically how do I define the proper namespaces/scope in the nanobind cpp file or cmake to enable such behavior of logically separating functions into different submodules exposed to python ?

Thank you !
